File: MissingInteger/ruby/MissingInteger.rb
class MissingInteger

  def solution(a)

    b = Array.new(a.length, 0)

    a.each_with_index do | value , i |
      b[value-1] = value if value >= 1 && value <= a.length
    end

    for i in 0..a.length
      
      return i + 1 if b[i] == 0

    end

    return a.length + 1

  end
  
end

File: MissingInteger/php/MissingInteger.php
<?php

namespace Codility;

class MissingInteger {

    function solution($A) {

        $B = array_fill(0, count($A), 0);

        foreach ( $A as $key => $value ) {

            if ( $value >= 1 && $value <= count( $A ) ) {
                $B[ $value - 1 ] = $value;
            }
            
        }

        for ( $i=0 ; $i <= count($A); $i++) {

            if ($B[$i] == 0) {
                return $i + 1;
            }

        }

        return  count($A)+1;

    }
}